src/esp/CMakeLists.txt

```diff
@@ -17,15 +17,595 @@ if(BUILD_WARNINGS_AS_ERRORS)
+  sensor/AudioSensor.h
+  sensor/AudioSensorStubs.h
+)
+
+if(BUILD_WITH_CUDA)
+  list(APPEND sensor_SOURCES sensor/RedwoodNoiseModel.cpp sensor/RedwoodNoiseModel.h)
+  # CUDA kernel sources
+  set(noise_model_SOURCES sensor/RedwoodNoiseModel.cu sensor/RedwoodNoiseModel.cuh)
+endif()
+
+#################
+## sim
+set(
+  sim_SOURCES
+  sim/AbstractReplayRenderer.cpp
+  sim/AbstractReplayRenderer.h
+  sim/BatchPlayerImplementation.cpp
+  sim/BatchPlayerImplementation.h
+  sim/BatchReplayRenderer.cpp
+  sim/BatchReplayRenderer.h
+  sim/ClassicReplayRenderer.cpp
+  sim/ClassicReplayRenderer.h
+  sim/Simulator.cpp
+  sim/Simulator.h
+  sim/SimulatorConfiguration.cpp
+  sim/SimulatorConfiguration.h
+)
+
+#################
+## build library
+
+################################
+# Keeping gfx_batch a separate library
 add_subdirectory(gfx_batch)
-add_subdirectory(assets)
-add_subdirectory(metadata)
-add_subdirectory(io)
-add_subdirectory(scene)
-add_subdirectory(physics)
-add_subdirectory(nav)
-add_subdirectory(sensor)
-add_subdirectory(sim)
+##############################
+
+set(habsim_SOURCES ${core_SOURCES})
+list(
+  APPEND
+  habsim_SOURCES
+  ${geo_SOURCES}
+  ${gfx_SOURCES}
+  ${assets_SOURCES}
+  ${metadata_SOURCES}
+  ${io_SOURCES}
+  ${scene_SOURCES}
+  ${physics_SOURCES}
+  ${nav_SOURCES}
+  ${sensor_SOURCES}
+  ${sim_SOURCES}
+)
+
+add_library(
+  habsim STATIC
+  ${habsim_SOURCES}
+)
+
+target_include_directories(
+  habsim PUBLIC ${PROJECT_BINARY_DIR}
+  PRIVATE "${DEPS_DIR}/tinyxml2" "${DEPS_DIR}/recastnavigation/Detour/Include"
+          "${DEPS_DIR}/recastnavigation/Recast/Include"
+)
+
+if(BUILD_WITH_CUDA)
+  add_library(noise_model_kernels STATIC ${noise_model_SOURCES})
+  target_link_libraries(noise_model_kernels PUBLIC ${CUDART_LIBRARY})
+  target_include_directories(
+    noise_model_kernels PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+  )
+  target_compile_features(noise_model_kernels PUBLIC cxx_std_11)
+
+  target_link_libraries(habsim PUBLIC noise_model_kernels ${CUDART_LIBRARY})
+
+  target_include_directories(habsim PRIVATE ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+endif()
+
+if(BUILD_WITH_AUDIO)
+  include_directories(${DEPS_DIR}/rlr-audio-propagation/RLRAudioPropagationPkg/headers)
+  target_link_libraries(habsim PUBLIC ${RLRAudioPropagation_LIBRARY})
+endif()
+
+target_link_libraries(
+  habsim
+  PUBLIC gfx_batch
+         Magnum::AnyImageImporter
+         Magnum::AnySceneImporter
+         Magnum::GL
+         Magnum::Magnum
+         Magnum::MaterialTools
+         Magnum::MeshTools
+         Magnum::DebugTools
+         Magnum::SceneGraph
+         Magnum::SceneTools
+         Magnum::Shaders
+         Magnum::Trade
+         Magnum::Primitives
+         MagnumPlugins::BasisImporter
+         MagnumPlugins::GltfImporter
+         MagnumPlugins::PrimitiveImporter
+         MagnumPlugins::StanfordImporter
+         MagnumPlugins::StbImageImporter
+         MagnumPlugins::StbImageConverter
```

Comment:
While you're doing this, could you also make the change to link the plugins (those, and all other `Importer`, `Converter` targets, also the ones linked conditionally below) only to shared libraries (i.e., the bindings module) and tests, and only where needed? E.g. the `IoTest` won't need any plugins.

Doing that is a significant speedup of the total build / link time.

Reply:
I'll ask you for specifics in slack if I need them :)